internal/cloudapi/v2/handler.go

```diff
@@ -176,6 +178,9 @@ func (h *apiHandlers) PostCompose(ctx echo.Context) error {
 
 	ctx.Logger().Infof("Job ID %s enqueued for operationID %s", id, ctx.Get(common.OperationIDKey))
 
+	// Save the request in the artifacts directory, ignore errors
+	_ = saveComposeRequest(h.server.workers.ArtifactsDir(), id, request)
```

Comment:
It's somewhat common :) I want the function to return errors so that they can potentially be caught or logged, but but since the storage of the request is not directly related to the success/failure of the compose I don't want a problem to block it from running.

Reply:
Thanks! My concern here is that if for some reason saveComposeRequest() always fails (programming error on our side, deployment issue, permission issue etc, disk-full) then this is never surfaced to the user as the error is ignored and there is no log, there is no error and the only way to notice is that the user will have an empty `Request` from ComposeMetadata. For this reason I think it might be worth logging it ?

(I personally would probably error if the saveComposeRequest() fails as it indicates something is wrong and log the error when readComposeRequest() fails (but continue as there is metadata just a reduced amount) and document in the API that request can be empty and that it means something with the metadata was wrong and let the client decide if it wants to error or continue with reduced data, but that is just me).

But as I said before, this is an outside perspective on osbuild-composer and I do not have a lot of context so it's fine to keep as is if that is preferred .

Reply:
Good point. Next problem is... There doesn't seem to be any kind of logging hooked up in the handler code. Not sure what to do about that.

Reply:
Thanks! I am by no means an expert here but it seems the pattern used is that the logger is part of the context, so something like:
```go
diff --git a/internal/cloudapi/v2/handler.go b/internal/cloudapi/v2/handler.go
index d507d4197..c94543ac9 100644
--- a/internal/cloudapi/v2/handler.go
+++ b/internal/cloudapi/v2/handler.go
@@ -179,7 +179,9 @@ func (h *apiHandlers) PostCompose(ctx echo.Context) error {
        ctx.Logger().Infof("Job ID %s enqueued for operationID %s", id, ctx.Get(common.OperationIDKey))
 
        // Save the request in the artifacts directory, ignore errors
-       _ = saveComposeRequest(h.server.workers.ArtifactsDir(), id, request)
+       if err := saveComposeRequest(h.server.workers.ArtifactsDir(), id, request); err != nil {
+               ctx.Logger().Warnf("Failed to save the component request: %v", err)
+       }
 
        return ctx.JSON(http.StatusCreated, &ComposeId{
                ObjectReference: ObjectReference{
@@ -606,7 +608,10 @@ func (h *apiHandlers) getComposeMetadataImpl(ctx echo.Context, id string) error
        }
 
        // Get the original compose request, if present
-       request, _ := readComposeRequest(h.server.workers.ArtifactsDir(), jobId)
+       request, err := readComposeRequest(h.server.workers.ArtifactsDir(), jobId)
+       if err != nil {
+               ctx.Logger().Warnf("Failed to read the component request: %v", err)
+       }
 
        if buildInfo.JobStatus.Finished.IsZero() {
                // job still running: empty response

```
should work if we want to go purely logging.